js/ui/tooltip-engine.js

```diff
@@ -188,166 +195,8 @@ window.TooltipEngine = Backbone.View.extend({
 
- /*
-  * This is the base that we build all of the tooltips on
-  *
-  * Every Tooltip has the following major parts:
-  * - initialize(), just accepts options and then tries to attach
-  *   the html for the tooltip by calling render() and bind() as required
-  *
-  * - render() and bind() to set up the HTML
-  *
-  * - A detector function. The detector functions are all bound to the
-  *   requestTooltip event in their respective bind() method. They receive an event with
-  *   information about where the cursor is and whether it got there because of a click,
-  *   selection character added, etc. It chooses to either load its tooltip or let the
-  *   event keep bubbling
-  *   > The detector function also sets aceLocation, which saves what portion of the
-  *     text the selector is active for.
-  *
-  * - updateText replaces whatever text is specified by the aceLocation
-  *   with the new text. It is common for tooltips to override this function
-  *   so that they can accept a value in a different format, make it into a string
-  *   and then pass the formatted value back to the function defined in TooltipBase
-  *   to do the actual replace
-  *
-  * - placeOnScreen which determines where the HTML needs to be moved to in order
-  *   for it to show up on by the cursor. This also pulls information from aceLocation
-  *
-  */
-
-window.TooltipBase = Backbone.View.extend({
-    bindToRequestTooltip: function() {
-        if (this.parent) {
-            this.callback = this.detector.bind(this);
-            this.parent.editor.on("requestTooltip", this.callback);
-        }
-    },
-
-    unbindFromRequestTooltip: function() {
-        if (this.parent) {
-            this.parent.editor.off("requestTooltip", this.callback);
-        }
-    },
-
-    placeOnScreen: function() {
-        // don't show tool tips when the editor is in readonly mode
-        if (this.parent.editor.getReadOnly()) {
-            return;
-        }
-        var parent = this.parent;
-        if (parent.currentTooltip && parent.currentTooltip !== this) {
-            parent.currentTooltip.$el.hide();
-        }
-        parent.currentTooltip = this;
-
-        var editor = parent.editor;
-        var loc = this.aceLocation;
-        var editorBB = editor.renderer.scroller.getBoundingClientRect();
-        var editorHeight = editorBB.height;
-        if (typeof loc.tooltipCursor !== "number") {
-            loc.tooltipCursor = loc.start + loc.length;
-        }
-        var coords = editor.renderer.textToScreenCoordinates(loc.row, loc.tooltipCursor);
-        var relativePos = coords.pageY - editorBB.top;
-
-        this.$el
-            .css({
-                top: $(window).scrollTop() + coords.pageY,
-                left: coords.pageX
-            })
-            .toggle(!(relativePos < 0 || relativePos >= editorHeight));
-    },
+TooltipEngine.registerTooltip = function(name, tooltipClass) {
+    TooltipEngine.classes[name] = tooltipClass;
+};
```

Comment:
You could also do:
```
export registerTooltip = (name, tooltipClass) => TooltipEngine.classes[name] = tooltipClass;
```
and then in files that use `registerTooltip` do:
```
import {registerTooltip} from "./tooltip-engine.js";
```
this is more idiomatic.